presto-main/src/main/java/com/facebook/presto/operator/scalar/ArraySortFunction.java

```diff
@@ -67,6 +67,17 @@ public Block sort(
             @Override
             public int compare(Integer p1, Integer p2)
             {
+                boolean nullLeft = block.isNull(p1);
+                boolean nullRight = block.isNull(p2);
+                if (nullLeft && nullRight) {
```

Comment:
I think it would be sufficient to just do:
```java
if (nullLeft) {
    return 1;
}

if (nullRight) {
    return -1;
}
```

Also, you don't need the "elses".